devicetypes/widomsrl/widom-smart-dry-contact.src/widom-smart-dry-contact.groovy

```diff
@@ -0,0 +1,548 @@
+            state "on", label: 'on', action: "switch.off", icon:"st.switches.switch.on", backgroundColor:"#00a0dc", nextState:"turningOff"
+        }
+        main "main"
+		details(["switch", "multiStatus"])
+
+	}
+
+    preferences {
+        input (
+                title: "WiDom Smart Dry Contact manual",
+                description: "Tap to view the manual.",
+                image: "https://www.widom.it/wp-content/uploads/2019/03/widom-3d-smart-dry-contact.gif",
+                url: "https://www.widom.it/wp-content/uploads/2020/04/Widom_Dry_Contact_IT_070420.pdf",
+                type: "href",
+                element: "href"
+        )
+
+        parameterMap().each {
+            input (
+                    title: "${it.num}. ${it.title}",
+                    description: it.descr,
+                    type: "paragraph",
+                    element: "paragraph"
+            )
+
+            input (
+                    name: it.key,
+                    title: null,
+                    //description: "Default: $it.def" ,
+                    type: it.type,
+                    options: it.options,
+                    //range: (it.min != null && it.max != null) ? "${it.min}..${it.max}" : null,
+                    defaultValue: it.def,
+                    required: false
+            )
+        }
+        input ( name: "logging", title: "Logging", type: "boolean", required: false )
+        section("CONFIGURE AN ASSOCIATION GROUP") { // CONFIGURE AN ASSOCIATION GROUP:
+        
+           input (
+                    type: "paragraph",
+                    element: "paragraph",
+                    title: "CONFIGURE ASSOCIATION GROUP:",
+                    description: "Association Group Members:\n" +
+                           "Enter a comma-delimited list of destinations (node IDs and/or endpoint IDs). " +
+                           "All IDs must be in hexadecimal format. E.g.:\n" +
+                           "Node destinations: '11, 0F'\n" +
+                           "Endpoint destinations: '1C:1, 1C:2'\n" +
+                           "To delete previous configurations you must enter 0 "
+                )
+            groupMap().each {
+                input (
+                    name: it.key,
+                    title: "${it.title} (${it.descr})",
+                    type: it.type,
+                    defaultValue: it.def,
+                    required: false
+                )
+            }
+		}
+    }
+}
+
+
+def on() {
+    encap(zwave.basicV1.basicSet(value: 255))
+}
+
+def off() {
+    encap(zwave.basicV1.basicSet(value: 0))
+}
+
+
+//Configuration and synchronization
+def updated() {
+    if ( state.lastUpdated && (now() - state.lastUpdated) < 500 ) return
+    def cmds = []
+    logging("Executing updated()","info")
+
+    if (device.currentValue("numberOfButtons") != 6) { sendEvent(name: "numberOfButtons", value: 6) }
```

Comment:
Since this DTH was based on fibaro-double-switch-2-zw5.groovy, please remove all the code that is related to Fibaro switch.

Reply:
Fixed